xen/arch/arm/p2m.c

```diff
@@ -1098,6 +1102,34 @@ int p2m_set_entry(struct p2m_domain *p2m,
         nr -= (1 << order);
     }
 
+    if ( likely(!rc) )
```

Comment:
Done. Updated comment in code:
"The reason to update IOMMU mapping outside "while loop" is that 
the IOMMU might not support the pages/superpages the CPU can deal
with (4K, 2M, 1G) and as result this will lead to non-optimal mapping.
Also we assume that the IOMMU mapping should be updated only
if CPU mapping passed successfully."